site/app/templates/officeHoursQueue/QueueStatus.twig

```diff
@@ -0,0 +1,117 @@
+    <input type="hidden" name="csrf_token" value="{{ csrf_token }}"/>
+    <label for="name_box">Your Name:</label>
+    <input type="text" id="name_box" name="name" maxlength="20" placeholder="Queue Name" value="{{viewer.getName()}}" style="margin:.5rem" required="required">
+    <br>
+    <div class="form-group">
+      <label for="queue_code">Queue Name:
+      </label>
+      <select id="queue_code" class="form-control" name="code" required="required" onchange="document.getElementById('add_to_queue').action = '{{base_url}}/'+this.value+'/add';">
+        <option value="">Queue Name</option>
+        {% for queue in viewer.getAllQueues() if queue['open'] == 'true' %}
+          <option value="{{queue['code']}}">{{viewer.removeUnderScores(queue['code'])}}</option>
+        {% endfor %}
+      </select>
+    </div>
+    <label for="token_box">Queue Access Code:</label>
+    <input type="text" name="token" id="token_box" placeholder="Queue Access Code" style="margin:.5rem" autocomplete="on" required="required">
+    <br>
+    {% if viewer.isContactInfoEnabled() %}
+      <label for="contact_info">My Contact Information (follow instructor directions):</label>
+      <input type="text" name="contact_info" id="contact_info" style="margin:.5rem" placeholder="Contact Information (follow instructor directions)" value="{{viewer.getContactInfo()}}" autocomplete="on" required="required">
+      <br>
+    {% endif %}
+    <button id="join_queue_btn" type="submit" class="btn btn-primary">Join</button>
+  </form>
+{% elseif viewer.getCurrentQueueStatus() == 'waiting' %}
+  <div id="queue_status">
+    {% if viewer.getQueueNumberAheadOfYou(viewer.getCurrentQueueCode()) - 1 == 1 %}
+      <p>There is currently
+        {{viewer.getQueueNumberAheadOfYou(viewer.getCurrentQueueCode()) - 1 }}
+        person ahead of you in the queue</p>
+    {% else %}
+      <p>There are currently
+        {{viewer.getQueueNumberAheadOfYou(viewer.getCurrentQueueCode()) - 1 }}
+        people ahead of you in the queue</p>
+    {% endif %}
+    <p>You are currently in:
+      <b>{{viewer.removeUnderScores(viewer.getCurrentQueueCode())}}</b>
+    </p>
+
+    <p>
+      {% if viewer.firstTimeInQueueThisWeek(viewer.getCurrentQueueLastHelped()) %}
+        Because you have not yet been helped through this queue during the last week, you may be prioritized.
+      {% elseif viewer.firstTimeInQueueToday(viewer.getCurrentQueueLastHelped()) %}
+        Because you have not yet been helped through this queue today, you may be prioritized.
+      {% else %}
+        However, because you have been helped through this queue today, you may be de-prioritized.
+      {% endif %}
+    </p>
+    <p>
+      {% set num_ahead_week = viewer.numberAheadInQueueThisWeek() %}
+      {% set num_ahead_today = viewer.numberAheadInQueueToday() %}
+      {% if viewer.firstTimeInQueueThisWeek(viewer.getCurrentQueueLastHelped()) %}
+        There
+        {{viewer.plurality(num_ahead_week, 'is', 'are')}}
```

Comment:
Let's make this a [twig function](https://twig.symfony.com/doc/3.x/advanced.html#functions) instead of a function off a model in-case anyone else wants to use this sort of functionality elsewhere in the site.